localstack/services/awslambda/lambda_utils.py

```diff
@@ -299,3 +300,106 @@ def generate_lambda_arn(
+            )
+    return True
+
+
+def verify_dict_filter(record_value: any, dict_filter: Dict[str, any]):
+    # https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax
+    fits_filter = False
+    for key, filter_value in dict_filter.items():
+        if key.lower() == "anything-but":
+            fits_filter = record_value not in filter_value
+        elif key.lower() == "numeric":
+            fits_filter = parse_and_apply_numeric_filter(record_value, filter_value)
+        elif key.lower() == "exists":
+            fits_filter = bool(filter_value)  # exists means that the key exists in the event record
+        elif key.lower() == "prefix":
+            if not isinstance(record_value, str):
+                LOG.warn(f"Record Value {record_value} does not seem to be a valid string.")
+            fits_filter = isinstance(record_value, str) and record_value.startswith(
+                str(filter_value)
+            )
+
+        if fits_filter:
+            return True
+    return fits_filter
+
+
+def filter_stream_record(filter_rule: Dict[str, any], record: Dict[str, any]):
+    if not filter_rule:
+        return True
+    # https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-syntax
+    filter_results = []
+    for key, value in filter_rule.items():
+        # check if rule exists in event
+        record_value = record.get(key.lower(), record.get(key))
+        append_record = False
+        if record_value is not None:
+            # check if filter rule value is a list (leaf of rule tree) or a dict (rescursively call function)
+            if isinstance(value, list):
+                if len(value) > 0:
+                    if isinstance(value[0], (str, int)):
+                        append_record = record_value in value
+                    if isinstance(value[0], dict):
+                        append_record = verify_dict_filter(record_value, value[0])
+                else:
+                    LOG.warn(f"Empty lambda filter: {key}")
+            if isinstance(value, dict):
+                append_record = filter_stream_record(value, record_value)
+        else:
+            # special case 'exists'
+            if isinstance(value, list) and len(value) > 0:
+                append_record = not value[0].get("exists", True)
+
+        filter_results.append(append_record)
+    return all(filter_results)
+
+
+def filter_stream_records(records, filters: List[FilterCriteria]):
+    filtered_records = []
+    for record in records:
+        for filter in filters:
+            for rule in filter["Filters"]:
+                if filter_stream_record(json.loads(rule["Pattern"]), record):
+                    filtered_records.append(record)
+                    break
+    return filtered_records
+
+
+def validate_filters(filter: FilterCriteria):
+    for rule in filter["Filters"]:
+        if not json.loads(rule["Pattern"]):
```

Comment:
Please add a try-except-block here, as some patterns might contain invalid json and will throw a `JSONDecodeError`. In this case we should also return `False`